loopchain/channel/channel_service.py

```diff
@@ -135,6 +138,37 @@ async def _serve():
             self._cleanup()
             loop.close()
 
+    async def _serve_manual_rollback(self):
+        """Initialize minimum channel resources and manual rollback
+
+        :return: None
+        """
+        await StubCollection().create_peer_stub()
+
+        results = await StubCollection().peer_stub.async_task().get_node_info_detail()
+        self._init_properties(**results)
+
+        self.__init_block_manager()
+        await self.__init_score_container()
+        await self.__inner_service.connect(conf.AMQP_CONNECTION_ATTEMPTS, conf.AMQP_RETRY_DELAY, exclusive=True)
+        await asyncio.sleep(0.01)   # sleep to complete peer service initialization
+
+        message = self._manual_rollback()
+        self.shutdown_peer(message=message)
+
+    def _manual_rollback(self) -> str:
+        logging.debug("_manual_rollback() start manual rollback")
+        if self.block_manager.blockchain.block_height >= 0:
+            self.block_manager.rebuild_block()
+
+        if self.block_manager.request_rollback():
+            message = "rollback finished"
+        else:
+            message = "rollback cancelled"
```

Comment:
Would it be better to add block height after rollback to message?
Is it possible to cancel rollback in progress?

Reply:
`loopchain` doesn't process rollback when failed rollback in `iconservice`.
In this case, show `rollback canceled` message.
It will show block height when rollback succeed.